dpe/src/dpe_instance.rs

```diff
@@ -202,6 +202,33 @@ impl<C: Crypto> DpeInstance<'_, C> {
 
         Ok(out_idx)
     }
+
+    pub(crate) fn add_tci_measurement(
+        &mut self,
+        idx: usize,
+        measurement: &TciMeasurement,
+    ) -> Result<(), DpeErrorCode> {
+        if idx >= MAX_HANDLES {
+            return Err(DpeErrorCode::InternalError);
+        }
+        let context = &mut self.contexts[idx];
```

Comment:
It feels a little bit error prone that we aren't checking locality, active status, etc. I know the expectation is that the caller will check this, I'm just worried this function encourages blindly adding a measurement.

Reply:
I could see how this could be abused if we weren't careful. The change did remind me there is a conversation I thought we should have about error handling.

I had originally placed the `add_tci_measurement` call higher because it was one of the last things that could fail before we made permanent changes to the state of the contexts. The system could have some undefined behaviors if failures happen late in any of the commands.

For example, in `DeriveChild` if the child is activated, added the parent, and then an error happens while adding the TCI measurement. It would return an error without cleaning anything up or undoing any of the partially executed work. There would now be a dangling context and nothing has the handle. This particular scenario might be really rare, but other may have the higher likelihoods.

We may want to copy out the context we are going to manipulate, work on it, and then commit it back to the DPE instance if everything passes. Do you think it is worth worrying about?

Reply:
As discussed offline, agreed this is a problem that we should fix. Might be good to open an issue to track.

Reply:
Added #63 to track the issue.